### PR TITLE
Remove `Result(minimize)` parameter [4/n]

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Part of #7183

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [n/a] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [n/a] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

**Review by ananthsub — COMMENTED** [review]
could we completely remove minimize from the result API? this is now.a very legacy holdover from TrainResult in pre-1.0 versions

**Comment by carmocca** [comment]
> could we completely remove minimize from the result API?

We still need a property/attribute/method to track the loss. This is probably going to be removed out of `Result` and into `ResultCollection` but we will still have something equivalent.

File: pytorch_lightning/core/step_result.py [hunk 1]
```diff
@@ -26,14 +26,8 @@
 
 class Result(Dict):
 
-    def __init__(self, minimize: Optional[Tensor] = None):
+    def __init__(self):
```
Comment (carmocca) [inline pytorch_lightning/core/step_result.py:29]:
```suggestion
    def __init__(self) -> None:
```

**Review by ananthsub — APPROVED** [review]

File: pytorch_lightning/core/step_result.py [hunk 2]
```diff
@@ -69,16 +63,18 @@ def __getstate__(self):
...
+    @property
+    def minimize(self) -> Optional[Tensor]:
+        return self.get('minimize', None)
+
+    @minimize.setter
+    def minimize(self, val: Optional[torch.Tensor]) -> None:
+        if val is not None:
+            if not isinstance(val, Tensor):
+                raise ValueError(f"`Result.minimize` must be a `torch.Tensor`, found: {val}")
+            if val.grad_fn is None:
```
Comment (tchaton) [inline pytorch_lightning/core/step_result.py:75]: Love this :)

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by SeanNaren — APPROVED** [review]

**Pull request merged**